File: scripts/Map/map.py
##############################################################################
# Folium Example
# ---------------------------------------------------------------------------
# Sources:
# https://github.com/python-visualization/folium
##############################################################################

import folium
import numpy as np
from folium.plugins import MarkerCluster

DIR = "/Users/sanchez.hmsc/Documents/GitHub/dataViz_CADi/scripts/Map/"

##############################################################################
# Basic globe with rastered map
##############################################################################
# Make an empty map
m = folium.Map(location=[20, 0], zoom_start=3.5)
m.save(DIR + 'map2.html')


m = folium.Map(location=[44, -73], zoom_start=5)
marker_cluster = MarkerCluster().add_to(m)

folium.Marker(
    location=[40.67, -73.94],
    popup='Add popup text here.',
    icon=folium.Icon(color='green', icon='ok-sign'),
).add_to(marker_cluster)

folium.Marker(
    location=[44.67, -73.94],
    popup='Add popup text here.',
    icon=folium.Icon(color='red', icon='remove-sign'),
).add_to(marker_cluster)

folium.Marker(
    location=[44.67, -71.94],
    popup='Add popup text here.',
    icon=None,
).add_to(marker_cluster)

m.save(DIR + 'manualClusteredMarkers.html')

##############################################################################
# Basic globe with rastered map
##############################################################################
size = 100
delta = .01
lats = np.random.uniform(37.773972-delta, 37.773972+delta, size=size)
lons = np.random.uniform(-122.43129-delta, -122.43129+delta, size=size)

locations = list(zip(lats, lons))
popups = ['lon:{}<br>lat:{}'.format(lon, lat) for (lat, lon) in locations]

m = folium.Map(
    location=[np.mean(lats), np.mean(lons)],
    tiles='Cartodb Positron',
    zoom_start=12
)

marker_cluster = MarkerCluster(
    locations=locations, popups=popups,
    name='1000 clustered icons',
    overlay=True,
    control=True
)

marker_cluster.add_to(m)
folium.LayerControl().add_to(m)

m.save(DIR + 'randomClusteredMarkers.html')

File: scripts/Map/map.R
##############################################################################
# Leaflet Example
# ---------------------------------------------------------------------------
# Sources:
# https://rstudio.github.io/leaflet/
# https://jakevdp.github.io/PythonDataScienceHandbook/04.13-geographic-data-with-basemap.html
##############################################################################

install.packages("rjson")
install.packages("dplyr")
#install.packages("ggplot2")
install.packages("rjson")
install.packages("RCurl")
install.packages("leaflet")

library(leaflet)
library(dplyr)
library(ggplot2)
library(rjson)
library(plotly)
library(jsonlite)
library(leaflet)
library(RCurl)

setwd("/Users/sanchez.hmsc/Documents/GitHub/dataViz_CADi/scripts/Map")

##############################################################################
# Basic globe with rastered map
##############################################################################
m = leaflet() %>% # Add default OpenStreetMap map tiles
  addTiles() %>% 
  addMarkers(lng=-99.226977, lat=19.595734, popup="New Zealand")
m
htmlwidgets::saveWidget(as_widget(m), "leafletMexico02.html")

##############################################################################

base_url = "http://data.colorado.gov/resource/j5pc-4t32.json?"
full_url = paste0(base_url, "station_status=Active", "&county=BOULDER")
water_data = getURL(URLencode(full_url))

# you can then pipe this
water_data_df = fromJSON(water_data) %>%
  flatten(recursive = TRUE) # remove the nested data frame

# turn columns to numeric and remove NA values
water_data_df = water_data_df %>%
  mutate_at(vars(amount, location.latitude, location.longitude), funs(as.numeric)) %>%
  filter(!is.na(location.latitude))
# create leaflet map
water_locations_map = leaflet(water_data_df) %>%
  addTiles() %>%
  addCircleMarkers(lng = ~location.longitude,
                   lat = ~location.latitude,
                   popup= "popup",
                   opacity = 0.75,
                   color = "#E83F6F",
                   stroke = FALSE,
                   radius=10
                   )
water_locations_map
htmlwidgets::saveWidget(as_widget(water_locations_map), "leafletColorado.html")
